File: others/comp9041-454/9041-labonserver/lab06/whale.pl
#!/usr/bin/perl -w
$name=$ARGV[0];
$line=0;
$totnum=0;
while ($data=<STDIN>){
   if ($data=~ /$name/){
     $line=$line+1;
     $currnum=(split / /,$data)[0];
     $totnum=$totnum+$currnum;
   }
}
printf "$name observations: $line pods, $totnum individuals\n";

File: others/comp9041-454/9041-labonserver/lab06/whale.py
#!/usr/bin/python
from sys import *
from re import *
name=argv[1]
line=0
totnum=0
while 1:
    try:
        data=raw_input()
        m=search(r'{}'.format(name),data)
        if m:
            line+=1
            currnum=data.split(' ')[0]
            totnum=totnum+int(currnum)
    except:
        print("{} observations: {} pods, {} individuals".format(name,line,totnum))
        exit()
